### ISSUE TITLE
"datalad addurls --version-urls" returns "NotImplementedError"

### ISSUE THREAD

**Issue by TobiasKadelka** [datalad/datalad#3840]
Hey,

while trying to addurls for creating are dataset with version urls, I get the following error:

```
[INFO] Versioned result for s3://hcp-openaccess/HCP_1200/100206/unprocessed/3T/tfMRI_WM_RL/LINKED_DATA/EPRIME/EVs/all_bk_err.txt?versionId=_Ipf_4BDKBcDxp_xaYRGnwZhI4eu0Ojn: s3://hcp-openaccess/HCP_1200/100206/unprocessed/3T/tfMRI_WM_RL/LINKED_DATA/EPRIME/EVs/all_bk_err.txt?versionId=_Ipf_4BDKBcDxp_xaYRGnwZhI4eu0Ojn 

[WARNING] error getting version of s3://hcp-openaccess/HCP_1200/100206/unprocessed/3T/tfMRI_WM_RL/LINKED_DATA/PHYSIO/100206_3T_tfMRI_WM_RL_Physio_log.txt?versionId=LzlIAb20H3RyD6QfAyLMudsk585JfE8Q: NotImplementedError() [s3.py:get_versioned_url:417] 
```

The command I ran:
datalad addurls --version-urls -c hcp_dataset -c inm7 code/tmp/$p.csv '{original_url}' '{subject}//{filename}'

From datalad help:
   -version-urls
              Try to add a version ID to the URL. This currently only has an effect on URLs for AWS S3 buckets. [Default: False]

**Comment by yarikoptic** [comment]
well, since URL is already versioned, no need for `--version-urls` in your case but I will push a "fix" to just tolerate such cases. 
The real problem for me is that even `datalad download-url s3://hcp-openaccess/HCP_900/100408/release-notes/Diffusion_unproc.txt?versionId=Ox8ZLNfErF0n7iGOVzwUEOQWLYs897F` doesn't seems to work (although I do not see immediate reason).  That is something to fix, trying to look into it now

**Comment by yarikoptic** [comment]
d'oh -- was missing a trailing '.' in the versionId.  With full one - seems to work just fine:
```
$> datalad --dbg download-url s3://hcp-openaccess/HCP_900/100408/release-notes/Diffusion_unproc.txt?versionId=Ox8ZLNfErF0n7iGOVzwUEOQWLYs897F.
[INFO   ] Downloading 's3://hcp-openaccess/HCP_900/100408/release-notes/Diffusion_unproc.txt?versionId=Ox8ZLNfErF0n7iGOVzwUEOQWLYs897F.' into '/home/yoh/proj/datalad/trash/hcp-tbl' 
[INFO   ] S3 session: Connecting to the bucket hcp-openaccess with authentication 
s3://hcp-openaccess/HCP_900/100408/release-notes/Diffusion_unproc.txt?versionId=Ox8ZLNfErF0n7iGOVzwUEOQWLYs897F.                                                                                                                download_url(ok): /home/yoh/proj/datalad/trash/hcp-tbl/Diffusion_unproc.txt (file)
add(ok): Diffusion_unproc.txt (file) [non-large file; adding content to git repository]
action summary:
  add (ok: 1)
  download_url (ok: 1)
  save (notneeded: 1)
```

**Comment by yarikoptic** [comment]
reconfirming that works within `addurls` (do not use `-c` which aren't in 0.11.x, so `datalad` special remote was initialized manually in the subdataset, and I guess it is the `hcp_dataset` which does it in your case)
```shell
$> datalad addurls hcp-tbl.csv '{original_url}' '{subject}//{filename}'           
[WARNING] Not creating subdataset at existing path: 100206 
addurl(ok): /home/yoh/proj/datalad/trash/hcp-tbl/100206/somelog.txt (file) [from datalad; to somelog.txt]                                                                                                                         
add(ok): 100206/somelog.txt (file)
metadata(ok): /home/yoh/proj/datalad/trash/hcp-tbl/100206/somelog.txt (file)                                                                                                                                                      
save(ok): /home/yoh/proj/datalad/trash/hcp-tbl/100206 (dataset)
save(ok): /home/yoh/proj/datalad/trash/hcp-tbl (dataset)
action summary:
  add (ok: 1)
  addurl (ok: 1)
  metadata (ok: 1)
  save (ok: 2)
(dev3) 1 3051.....................................:Tue 29 Oct 2019 02:45:44 PM EDT:.
(git-annex)hopa:~datalad/trash/hcp-tbl[master]git-annex
$> cat hcp-tbl.csv 
original_url,subject,filename
s3://hcp-openaccess/HCP_1200/100206/unprocessed/3T/tfMRI_WM_RL/LINKED_DATA/PHYSIO/100206_3T_tfMRI_WM_RL_Physio_log.txt?versionId=LzlIAb20H3RyD6QfAyLMudsk585JfE8Q,100206,somelog.txt

$> datalad --version
datalad 0.11.8.dev8
```

**Comment by kyleam** [comment]
gh-3842 was tagged as closing this, and those changes made their way into master with 56c1c9f18 (Merge remote-tracking branch 'origin/0.11.x', 2019-10-29).  This issue wasn't auto-closed because that series was based off of 0.11.x and didn't contain a commit with a closes/fixes cookie.
